File: lab summer25/assignment5/taskB.py
import sys
# from collections import deque

sys.setrecursionlimit(2*100000+5)
sys.stdin = open("input.txt","r")

# def dfs(graph,n):
#     visited = [False]*(n+1)

#     for start in range(1,n+1):
#         if visited[start]: continue
#         stack = deque()
#         stack.append(start)

#         while stack:
#             at = stack[-1]
#             if not visited[at]: print(at,end=' ')
#             visited[at] = True
#             found = False
#             for v in graph[at]:
#                 if visited[v]: continue
#                 found = True
#                 stack.append(v)
#                 break
#             if not found: stack.pop()

def dfs(graph,root,visited):
    visited[root] = True
    print(root , end=' ')
    for v in graph[root]:
        if not visited[v]: dfs(graph,v,visited)

def solve():
    n,m = list(map(int,input().split()))
    u = list(map(int,input().split()))
    v = list(map(int,input().split()))
    graph = [[] for _ in range(n+1)]
    for i in range(m):
        graph[u[i]].append(v[i])
        graph[v[i]].append(u[i])
    visited = [False]*(n+1)
    for v in range(1,n+1):
        if not visited[v]: dfs(graph,v,visited)

t = 1
t = int(input())
while t>0:
    t-=1
    solve()
    print()

File: lab spring25/assignment2/taskB.cpp
#include <stdio.h>
#define br printf("\n")

void solve(){
    int n,m;
    scanf("%d",&n);
    int a1[n];
    for(int i=0;i<n;i++) scanf("%d",&a1[i]);
    scanf("%d",&m);
    int a2[m];
    for(int i=0;i<m;i++) scanf("%d",&a2[i]);

    int arr[n+m],at=0,i=0,j=0;
    while (i<n && j<m)
    {
        if(a1[i]<a2[j]){
            printf("%d ",a1[i++]);
        }else{
            printf("%d ",a2[j++]);
        }
    }

    for(i;i<n;i++) printf("%d ",a1[i]);
    for(j;j<m;j++) printf("%d ",a2[j]);
    br;
}

int main(){
    freopen("input.txt","r",stdin);
    solve();
}